src/puppetlabs/trapperkeeper/authorization/ring_middleware.clj

```diff
@@ -209,11 +209,14 @@
    allow-header-cert-info :- schema/Bool
    oid-map :- acl/OIDMap]
   (if-let [cert (request->cert request allow-header-cert-info)]
-    (let [extensions (ssl-utils/get-extensions cert)
+    (let [oid-map (merge acl/default-oid-map oid-map)
+          extensions (ssl-utils/get-extensions cert)
           translate-oids (fn [out extension]
                            (let [oid-key (get oid-map (:oid extension)
                                               (keyword (:oid extension)))
-                                 value (str (:value extension))]
+                                 value (if (= :subjectAlternativeName oid-key)
+                                         (ssl-utils/get-subject-dns-alt-names cert)
```

Comment:
With respect to the matching that we're doing where a 'certname' value for an ACE could match a Subject CN or a dNSName entry in the Subject Alternative Name extension, I think this makes sense.

From https://tools.ietf.org/html/rfc2818#section-3.1:

```
If a subjectAltName extension of type dNSName is present, that MUST
be used as the identity. Otherwise, the (most specific) Common Name
field in the Subject field of the certificate MUST be used. Although
the use of the Common Name is existing practice, it is deprecated and
Certification Authorities are encouraged to use the dNSName instead.
```

Arguably the above could be interpreted to mean we should only match from a dNSName entry - and not the Subject CN - if both are present in a certificate.  I don't think we can make that leap, though, for compatibility reasons, so I'm good with what you have here from that perspective.

However, I'm thinking it may be better for an ACE that expresses a SubjectAltName constraint to not forceably limit the matches to dNSName entries only.  For example, I think it might be better to allow for it to match per any of the available GeneralName types - dNSName, iPAddress, ...

From that perspective, it might actually be better for rules with SubjAltName ACEs to be able to have the match include the GeneralName type.  So, for example, you could express a rule in your TK config that would have something like:

````
allow: {
  extensions: {
    subject_alt_name: 
       {  dns_name: [ "myhost1", "myhost2" ],
          ip_addr: [ "1.2.3.4" ]
       }
    }
  }
```

I'd be fine even with not implementing support for anything but "dns_name" as a GeneralName type for now.  If we require the names to be qualified with "dns_name" at the rule level for now, though, that would give us the ability to expand to support other GeneralName types later when/if needed.

Does this make sense?